eos_data_distribution/ndn/manifest.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
+# -*- Mode:python; coding: utf-8; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+#
+# Copyright (C) 2016 Endless Computers INC.
+# Author: Niv Sardi <xaiki@endlessm.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# A copy of the GNU Lesser General Public License is in the file COPYING.
+
+import logging
+
+from pyndn import Name
+
+from . import http
+from . import chunks
+
+logger = logging.getLogger(__name__)
+
+# the manifest producer is an http Producer that answers on a different name
+class Producer(chunks.Producer):
+    def __init__(self, name, url, session=None, *args, **kwargs):
+        self._getter = http.Getter(url, session)
+        self._getter.connect('data', lambda o, d: self.sendFinish(d))
+
+        # XXX -- we mangle the name in the constructor, this is slow
+        self._reply_name = Name(name).append(self._getter._last_modified)
+
+        super(Producer, self).__init__(name, *args, **kwargs)
```

Comment:
because in this case we need to reply on another name (A lower name) than the one we own.